app/src/main/java/com/rootstrap/android/ui/compose_navigation/NavigationRoutes.kt

```diff
@@ -2,10 +2,19 @@ package com.rootstrap.android.ui.compose_navigation
 
 import com.rootstrap.android.R
 
-sealed class NavigationRoutes(val route: String, var title: String, var icon: Int?) {
-    object Login : NavigationRoutes("login", "", null)
-    object ProductsList : NavigationRoutes("products_list", "", null)
+sealed class NavigationRoutes(
```

Comment:
I'm not getting why using a `sealed class` with `objects` insde.
It appears to be more an `enum class`.
Which is the purpose of the choice you made?

Reply:
One of the advantages of the sealed classes vs enum is in when statements, it's safer since in compilation time checks that all branches are being handled. Another thing is that it has more flexibility, I can add a data class in the future is it's required.

Reply:
I know some of the differences between `sealed` and `enum`,
but its strange to me to define the different type of the sealed class inside it (I understood that)
Is there any other reviewer which can help us to define this? 😅

Reply:
This is the recommended way to do it,
as @agustinkoll-rootstrap says the sealed class has more flexibility to add data than enums, 
i.e:
```
object ProductsDetails : NavigationRoutes("products_details/{id}") {
       fun routeForProduct(id: String) = "products_details/$id"
}
```
or if you want to use a data class to define custom data like:
`data class ProductsList(@DrawableRes customIcon: Int? = null) : NavigationRoutes("products_list", "", customIcon)`
and more
as you see, an enum is very limited
and this is the recommended way by [Google](https://developer.android.com/jetpack/compose/navigation)